handlers/product-switch-api/test/contributionToSupporterPlusIntegration.test.ts

```diff
@@ -126,15 +129,20 @@ describe('product-switching behaviour', () => {
 	it(
 		'can adjust an invoice to zero',
 		async () => {
-			// To run this test you will need to find a recurring contribution which has been switched to supporter plus but payment hasn't been taken
-			const zuoraClient = await ZuoraClient.create('CODE');
-			const invoiceId = '8ad093788f0f4d2b018f105fe4357ff1';
+			const contributionPrice = 2;
+			const { zuoraClient, switchInformation } = await createTestContribution(
+				contributionPrice,
+				2.1,
+				false,
+			);
+
+			const switchResponse = await doSwitch(zuoraClient, switchInformation);
 
 			const response = await adjustNonCollectedInvoice(
 				zuoraClient,
-				invoiceId,
-				1.33,
-				'8ad08cbd8586721c01858804e3715378',
+				switchResponse.invoiceIds?.[0] ?? '',
```

Comment:
I'd rather not go ahead with the call if the value isn't present
```suggestion
      const invoiceId = getIfDefined(
        switchResponse.invoiceIds?.[0],
        'invoice id was undefined in response from Zuora',
	    );
      
			const response = await adjustNonCollectedInvoice(
				zuoraClient,
				invoiceId,
```